plugins/url/reddit.go

```diff
@@ -45,37 +45,63 @@ type redditComment struct {
 	} `json:"data"`
 }
 
-var redditUserRegex = regexp.MustCompile(`^/(u|user)/([^/]+)$`)
+var redditUserRegex = regexp.MustCompile(`^/(?:u|user)/([^\s/]+)$`)
+var redditPrivmsgUserRegex = regexp.MustCompile(`(?:\s|^)/(?:u|user)/([^\s/]+)`)
 var redditCommentRegex = regexp.MustCompile(`^/r/[^/]+/comments/([^/]+)/.*$`)
-var redditSubRegex = regexp.MustCompile(`^/r/([^/]+)/?.*$`)
+var redditSubRegex = regexp.MustCompile(`^/r/([^\s/]+)/?.*$`)
+var redditPrivmsgSubRegex = regexp.MustCompile(`(?:\s|^)/r/([^\s/]+)`)
 var redditPrefix = "[Reddit]"
 
-func newRedditProvider(urlPlugin *Plugin) {
+func newRedditProvider(m *seabird.BasicMux, urlPlugin *Plugin) {
+	m.Event("PRIVMSG", redditPrivmsgCallback)
 	urlPlugin.RegisterProvider("reddit.com", redditCallback)
 }
 
+func redditPrivmsgCallback(b *seabird.Bot, r *seabird.Request) {
+	content := r.Message.Trailing()
+
+	for _, matches := range redditPrivmsgSubRegex.FindAllStringSubmatch(content, -1) {
+		redditGetSub(b, r, matches[1])
+	}
+
+	for _, matches := range redditPrivmsgUserRegex.FindAllStringSubmatch(content, -1) {
+		redditGetUser(b, r, matches[1])
+	}
+}
+
 func redditCallback(b *seabird.Bot, r *seabird.Request, u *url.URL) bool {
 	text := u.Path
+
 	//nolint:gocritic
 	if redditUserRegex.MatchString(text) {
-		return redditGetUser(b, r, text)
+		matches := redditUserRegex.FindStringSubmatch(text)
```

Comment:
It might be a little bit clearer to do something like the following to put it all in one line:

```golang
if matches := redditUserRegex.FindStringSubmatch(text) {
```

Same for the instances below